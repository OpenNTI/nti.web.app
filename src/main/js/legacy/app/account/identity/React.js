const Ext = require('@nti/extjs');
const { getService } = require('@nti/web-client');
const { Identity } = require('@nti/web-profiles');
const LoginActions = require('internal/legacy/login/Actions');
require('internal/legacy/overrides/ReactHarness');

const AccountActions = require('../Actions');

module.exports = exports = Ext.define(
	'NextThought.app.account.identity.React',
	{
		extend: 'NextThought.ReactHarness',
		alias: 'widget.identity-react',
		component: Identity.Session,

		setTheme(scope, knockout) {},

		initComponent: function () {
			this.callParent(arguments);
			Object.assign(this.initialConfig, {
				addHistory: true,
				baseroute: '/app',
			});
			this.AccountActions = AccountActions.create();
			this.LoginActions = LoginActions.create();
		},

		async afterRender() {
			this.callParent(arguments);
			const service = await getService();
			const supportLinks = service.getSupportLinks();

			this.mon(Ext.getBody(), {
				click: this.onBodyClicks.bind(this, supportLinks),
			});
		},

		onBodyClicks(supportLinks, e) {
			let action = e.getTarget('a[data-action]')?.dataset.action;
			const link = e.getTarget('a[href]');
			const href = link?.getAttribute('href');

			// This is used to launch support logic the same way everywhere
			// we have a link that goes to the supportContact url. (so we can
			// just emit that url in the dom and not worry about coping logic)
			if (
				supportLinks.supportContact === href &&
				supportLinks.internalSupport
			) {
				action = action || 'support';
			}

			if (action) {
				if (call(this, `do-${action}`, link, e) !== false) {
					e.stopEvent();
				}
			}
		},

		showLink(target) {
			target.setAttribute('target', '_blank');
			return false;
		},

		'do-welcomeGuide'(target) {
			this.AccountActions.showWelcomePage(target.href);
		},

		'do-about': 'showLink',

		'do-privacy'(target) {
			this.AccountActions.showPrivacy(target);
		},

		'do-privacyForMinors'(target) {
			this.AccountActions.showChildrensPrivacy(target.href);
		},

		'do-terms'(target) {
			this.AccountActions.showTermsOfService(target.href);
		},

		'do-support'(target) {
			this.AccountActions.showContactUs();
		},

		'do-helpSite': 'showLink',

		'do-impersonate'() {
			this.LoginActions.handleImpersonate();
		},

		'do-logout'() {
			this.LoginActions.handleLogout();
		},
	}
);

function call(scope, action, ...args) {
	if (typeof action === 'string') {
		return call(scope, scope[action], ...args);
	}

	return action.apply(scope, args);
}
