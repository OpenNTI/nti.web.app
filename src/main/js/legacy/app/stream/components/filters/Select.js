const Ext = require('extjs');
const Select = require('legacy/common/form/fields/select');

module.exports = exports = Ext.define('NextThought.app.stream.components.filters.Select', {
	extend: 'Ext.Component',
	alias: 'widget.stream-filter-select',

	cls: 'group',

	renderTpl: Ext.DomHelper.markup([
		{cls: 'name', html: '{name}'}
	]),

	initComponent () {
		this.callParent(arguments);

		if (this.typeCls) {
			this.addCls(this.typeCls);
		}


		this.valueMap = this.items.reduce((acc, item) => {
			if (item.stateValue) {
				acc[item.stateValue] = item.value;
			}

			return acc;
		}, {});
	},


	beforeRender () {
		this.callParent(arguments);

		this.renderData = Ext.apply(this.renderData || {}, {
			name: this.displayText
		});
	},


	afterRender () {
		this.callParent(arguments);

		this.setUpInputs();
	},


	setUpInputs () {
		this.selectCmp = Select.create({
			renderTo: this.el,
			options: this.items.map((item) => {
				return {
					displayText: item.displayText,
					value: item.stateValue || item.state
				};
			}),
			onChange: this.onInputChange.bind(this)
		});

		this.on('destroy', () => { this.selectCmp.destroy(); });
	},


	onInputChange () {
		if (this.onChange) {
			this.onChange();
		}
	},


	setValue (value) {
		this.selectCmp.selectValue(value);
	},


	getValue () {
		let value = this.selectCmp.getSelectedValue();

		return value;
	},


	getParamValue (value) {
		return this.valueMap.hasOwnProperty(value) ? this.valueMap[value] : value;
	},


	getParamFromState (state) {
		return this.getParamValue(state);
	}
});
