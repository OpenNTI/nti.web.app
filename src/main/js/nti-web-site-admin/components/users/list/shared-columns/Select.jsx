import { useCallback } from 'react';

import { Input, Variant } from '@nti/web-core';
import { useProperties, StateStore } from '@nti/web-core/data';

import { Centered } from './Common';

const { Selectable } = StateStore.Behaviors;

const Cell = styled('td')`
	&.selected {
		&,
		& ~ td {
			background-color: var(--color-selected-background-light);
		}
	}
`;

SelectColumn.HeaderPlaceholderComponent = () => (
	<Input.Checkbox p="sm" disabled />
);
SelectColumn.HeaderComponent = () => {
	const { isAllSelected, selectAll, deselectAll } = useProperties(
		Selectable.hasBehavior
	);

	const onChange = useCallback(
		e => {
			if (isAllSelected()) {
				deselectAll();
			} else {
				selectAll();
			}
		},
		[isAllSelected, selectAll, deselectAll]
	);

	return (
		<Input.Checkbox checked={isAllSelected()} onChange={onChange} p="sm" />
	);
};

SelectColumn.CSSClassName = css`
	width: 70px;
`;

SelectColumn.RendersContainer = true;
SelectColumn.Placeholder = () => null;
SelectColumn.Create = props => Variant(SelectColumn, props);
export function SelectColumn({ item }) {
	const { isSelected, select, deselect } = useProperties(
		Selectable.hasBehavior
	);

	const selected = isSelected(item);
	const onChange = useCallback(
		e => {
			if (selected) {
				deselect(item);
			} else {
				select(item);
			}
		},
		[item, select, deselect, selected]
	);

	return (
		<Cell selected={selected}>
			<Centered>
				<Input.Checkbox checked={selected} onChange={onChange} />
			</Centered>
		</Cell>
	);
}
