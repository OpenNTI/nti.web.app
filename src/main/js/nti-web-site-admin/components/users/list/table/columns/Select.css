.select-column {
	width: 70px;
}

.row-selected {
	&,
	& ~ td {
		background-color: rgba(63, 179, 246, 0.05);
	}
}

.cell {
	composes: veridically-center from './column.css';
	align-items: center;
}
