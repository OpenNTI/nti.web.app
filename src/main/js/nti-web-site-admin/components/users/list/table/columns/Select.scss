.users-table-container {
	.select-col {
		width: 70px;
	}

	tbody {
		.select-col {
			.checkbox-component {
				padding-left: 30px;
				position: relative;
				top: 20px;
			}

			&.row-selected {
				&, & ~ td {
					background-color: rgba(63, 179, 246, 0.05);
				}
			}
		}
	}
}
