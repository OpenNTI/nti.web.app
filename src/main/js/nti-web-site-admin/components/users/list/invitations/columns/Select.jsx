import React, { useCallback } from 'react';

import { Input } from '@nti/web-core';

import { InvitationsStore } from '../Store';

import { Centered } from './Common';

const Cell = styled('td')`
	&.selected {
		&,
		& ~ td {
			background-color: var(--color-selected-background-light);
		}
	}
`;

Select.HeaderPlaceholderComponent = () => <Input.Checkbox p="sm" disabled />;
Select.HeaderComponent = () => {
	const { isAllSelected, selectAll, deselectAll } =
		InvitationsStore.useProperties();

	const onChange = useCallback(
		e => {
			if (isAllSelected()) {
				deselectAll();
			} else {
				selectAll();
			}
		},
		[isAllSelected, selectAll, deselectAll]
	);

	return (
		<Input.Checkbox checked={isAllSelected()} onChange={onChange} p="sm" />
	);
};

Select.RendersContainer = true;
Select.CSSClassName = css`
	width: 70px;
`;

Select.Placeholder = () => null;
export function Select({ item }) {
	const { isSelected, select, deselect } = InvitationsStore.useProperties();

	const selected = isSelected(item);
	const onChange = useCallback(
		e => {
			if (selected) {
				deselect(item);
			} else {
				select(item);
			}
		},
		[item, select, deselect, selected]
	);

	return (
		<Cell selected={selected}>
			<Centered>
				<Input.Checkbox checked={selected} onChange={onChange} />
			</Centered>
		</Cell>
	);
}
