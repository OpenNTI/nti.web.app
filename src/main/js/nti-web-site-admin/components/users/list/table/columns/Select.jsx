import React from 'react';
import PropTypes from 'prop-types';
import {Checkbox} from  '@nti/web-commons';
import {Connectors} from '@nti/lib-store';
import cx from 'classnames';

@Connectors.Any.connect({
	isAllSelected: 'isAllSelected',
	selectAll: 'selectAll',
	deselectAll: 'deselectAll'
})
class SelectHeader extends React.Component {
	static propTypes = {
		isAllSelected: PropTypes.func,
		selectAll: PropTypes.func,
		deselectAll: PropTypes.func
	}

	render () {
		const {isAllSelected, selectAll, deselectAll} = this.props;

		return (
			<Checkbox
				checked={isAllSelected()}
				onChange={(e) => {
					if(e.target.checked) {
						selectAll();
					}
					else {
						deselectAll();
					}
				}
				}/>
		);
	}
}

export default
@Connectors.Any.connect({
	onSelect: 'onSelect',
	onDeselect: 'onDeselect',
	isSelected: 'isSelected'
})
class Select extends React.Component {
	static propTypes = {
		item: PropTypes.object.isRequired,
		onSelect: PropTypes.func,
		onDeselect: PropTypes.func,
		isSelected: PropTypes.func
	}

	static rendersContainer = true;

	static HeaderComponent = SelectHeader

	static cssClassName = 'select-col';

	onSelectChanged = (e) => {
		const {item, onSelect, onDeselect} = this.props;

		if(e.target.checked) {
			onSelect(item);
		}
		else {
			onDeselect(item);
		}
	}

	render () {
		const {item, isSelected} = this.props;

		const isItemSelected = isSelected(item);

		return (
			<td className={cx('select-col', {'row-selected': isItemSelected})}>
				<div className={cx('cell')}>
					<Checkbox checked={isItemSelected} onChange={this.onSelectChanged}/>
				</div>
			</td>
		);
	}
}
