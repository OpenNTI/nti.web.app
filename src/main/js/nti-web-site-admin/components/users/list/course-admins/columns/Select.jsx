import { SelectColumn } from '../../shared-columns/Select';

export const Select = SelectColumn;
