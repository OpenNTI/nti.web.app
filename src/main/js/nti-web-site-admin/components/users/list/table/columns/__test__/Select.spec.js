/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';

import Select from '../Select';

describe('Site admin user table select column test', () => {
	test('Test not selected', () => {
		const colCmp = renderer.create(<Select
			isSelected={(item)=>{
				return item.id === 'selectedUser';
			}}
			item={{
				id: 'unselectedUser'
			}}/>);

		const tree = colCmp.toJSON();

		// should have a checked=false checkbox and no row-selected class
		expect(tree).toMatchSnapshot();
	});

	test('Test selected', () => {
		const colCmp = renderer.create(<Select
			isSelected={(item)=>{
				return item.id === 'selectedUser';
			}}
			item={{
				id: 'selectedUser'
			}}/>);

		const tree = colCmp.toJSON();

		// should have a checked=true checkbox and a row-selected class
		expect(tree).toMatchSnapshot();
	});
});
