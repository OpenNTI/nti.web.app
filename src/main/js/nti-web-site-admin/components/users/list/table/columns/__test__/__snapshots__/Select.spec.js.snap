// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Site admin user table select column test Test not selected 1`] = `
<td
  className="select-col"
>
  <div
    className="cell"
  >
    <label
      className="checkbox-component"
    >
      <input
        checked={false}
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="label"
      />
    </label>
  </div>
</td>
`;

exports[`Site admin user table select column test Test selected 1`] = `
<td
  className="select-col row-selected"
>
  <div
    className="cell"
  >
    <label
      className="checkbox-component"
    >
      <input
        checked={true}
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="label"
      />
    </label>
  </div>
</td>
`;
